core/src/test/java/org/testcontainers/junit/DockerfileTest.java

```diff
@@ -0,0 +1,74 @@
+package org.testcontainers.junit;
+
+import com.github.dockerjava.api.command.BuildImageCmd;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.pass;
+
+public class DockerfileTest {
```

Comment:
This is interesting and good stuff. If I understand this right there are three ways of providing a 'Dockerfile' - an actual classpath file, as a String, and via the builder pattern?

That's more than I was expecting but certainly more flexible!